### PR TITLE
FOUR-7495 Canvas Selection

### PR DESCRIPTION
**Pull request by rodriquelca**
## Issue & Reproduction Steps

Expected behavior: 
Canvas Selection
Actual behavior: 
The modeler has not a Canvas Selection feature
## Solution
- Canvas Selection was improved.
- Cypress tests was added.
- First crown for selector was added.
- Suppor drag.
- Multiselector.
- shift key support.

## How to Test
Go to Modeler project and use the selector feature in order to provide more ux experience.

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-7495

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

### CODE REVIEW

File: src/components/crown/crownMultiselect/crownMultiselect.vue [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  created() {
+    this.$t = this.$t.bind(this);
+  },
+  computed: {
+    isMultiSelect() {
+      const countSelected = store.getters.highlightedShapes.length;
+      return countSelected > 1;
+    },
+    isFlow() {
+      return false;
```
Comment (josechirivella) [inline src/components/crown/crownMultiselect/crownMultiselect.vue:71]: this should be a data() property is always is going to be returning false

File: src/components/crown/crownMultiselect/crownMultiselect.vue [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    },
+    highlightedShapes: () => store.getters.highlightedShapes,
+  },
+  methods: {
+    copyElement() {
+      // @todo: Implement copyElement
```
Comment (josechirivella) [inline src/components/crown/crownMultiselect/crownMultiselect.vue:77]: remove the todo if completed

File: src/components/crown/crownMultiselect/crownMultiselect.vue [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      // eslint-disable-next-line no-console
+      console.log('@todo: Implement copyElement');
+      this.$emit('copy-nodes');
+    },
+    deleteElement() {
+      // @todo: Implement deleteElement
```
Comment (josechirivella) [inline src/components/crown/crownMultiselect/crownMultiselect.vue:83]: remove the todo

File: src/components/crown/crownMultiselect/crownMultiselect.vue [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+      );
+      this.savePositionOnPointerupEventSet = false;
+
+      if (!store.getters.allowSavingElementPosition) {
+        return;
+      }
+
+      this.$emit('save-state');
+    },
+    repositionCrown() {},
```
Comment (josechirivella) [inline src/components/crown/crownMultiselect/crownMultiselect.vue:105]: remove empty function

File: src/components/crown/crownMultiselect/crownMultiselect.vue [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+      );
+    },
+    setUpPositionHandling() {},
+  },
+  async mounted() {
+    await this.$nextTick();
+    await this.paperDoneRendering();
+    this.setUpCrownConfig();
+  },
+  destroyed() {},
```
Comment (josechirivella) [inline src/components/crown/crownMultiselect/crownMultiselect.vue:126]: remove

**Review by josechirivella — CHANGES_REQUESTED** [review]

File: src/components/crown/crownMultiselect/crownMultiselect.vue (hunk 2 above)
Comment (caleeli) [reply src/components/crown/crownMultiselect/crownMultiselect.vue:77]: This will be implemented in the copy/paste ticket.

**Review by caleeli — APPROVED** [review]

File: src/components/crown/crownMultiselect/crownMultiselect.vue [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+      const countSelected = store.getters.highlightedShapes.length;
+      return countSelected > 1;
+    },
+    highlightedShapes: () => store.getters.highlightedShapes,
+  },
+  methods: {
+    copyElement() {
+      // @todo: Implement copyElement
+      // eslint-disable-next-line no-console
+      console.log('@todo: Implement copyElement');
```
Comment (TinCodes) [inline src/components/crown/crownMultiselect/crownMultiselect.vue:76]: Remove `console.log()` statements please

File: src/components/modeler/setUpSelectionBox.js [hunk 7]
```diff
@@ -100,7 +100,7 @@ export default function setUpSelectionBox(setCursor, resetCursor, paperManager,
       .map(shape => shape.model.component.node);
 
     store.commit('addToHighlightedNodes', selectedNodes);
-
+
```
Comment (TinCodes) [inline src/components/modeler/setUpSelectionBox.js:103]: Extra tabulation

File: tests/e2e/support/utils.js [hunk 8]
```diff
@@ -39,13 +39,13 @@ export function getGraphElements() {
...
-export function getElementAtPosition(position, componentType) {
+export function getElementAtPosition(position, componentType, offsetX = 0, offsetY = 0) {
   const paperGridSize = gridSize;
   const searchRectangle = {
     width: paperGridSize,
     height: paperGridSize,
-    x: position.x - paperGridSize / 2,
-    y: position.y - paperGridSize / 2,
+    x: position.x - paperGridSize / 2 + offsetX,
+    y: position.y - paperGridSize / 2 + offsetY,
```
Comment (TinCodes) [inline tests/e2e/support/utils.js:48]: As a suggestion: Use () to group the desired expressions and avoid unwanted calculations

**Pull request merged**